### PR TITLE
Several improvements to ManagedHandler

### PR DESCRIPTION
**Pull request by stephentoub**
The main change is to switch how reading is done so that the core logic is handed back a line at a time to parse rather than having to parse byte-by-byte.  I also separated classes out into their own files to make the solution a bit more manageable, and employed several others perf tweaks, like using a single empty response stream when possible, using Span instead of ArraySegment where possible, etc.  For simple get requests, this improves throughput by 5-10%.

cc: @geoffkizer

### CODE REVIEW

File: src/System.Net.Http/src/System/Net/Http/Managed/ValueStringBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text
+{
+    internal struct ValueStringBuilder
```
Comment (danmosemsft) [inline src/System.Net.Http/src/System/Net/Http/Managed/ValueStringBuilder.cs]: Do we not have any generic char array builder type that could do this job? It seems very generic.

Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Http/Managed/ValueStringBuilder.cs]: If we do, please point me to it.

Comment (danmosemsft) [inline src/System.Net.Http/src/System/Net/Http/Managed/ValueStringBuilder.cs]: I guess the nearest thing is `List<T>` but the code is a little more convoluted, it has an extra `version` field, and there is no efficient way to get a string out of it...

Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Http/Managed/ValueStringBuilder.cs]: And it's a class.

**Comment by stephentoub** [comment]
@dotnet-bot test UWP CoreCLR x64 Debug Build please ("Internal.Cryptography.CryptoThrowHelper+WindowsCryptographicException : Unknown error (0xc0000001)" in AsymmetricRSASignature test)

**Pull request merged**